### ISSUE TITLE
Plugin private builds into Core-Setup and CLI

### ISSUE THREAD

**Issue by gkhanna79** [dotnet/core-setup#118]
For basic E2E validation, currently core-setup needs to consume packages (CoreCLR, CoreFX) to create NETCore.App metapackage and then CLI consumes it to produce the CLI redists/packages - and this is only supported via automation, post commit.

There is no way for a developer to validate if their change will introduce a regression in core-setup or CLI. For example, if one were to build a private CoreCLR package, on a particular OS (e.g. Windows) and want to test it through, there are no extensibility points to do so.

It is imperative to have this in order to identify issues sooner than later.

CC @piotrpMSFT @brthor @eerhardt @schellap @markwilkie @weshaggard

**Comment by ellismg** [comment]
> There is no way for a developer to validate if their change will introduce a regression in core-setup or CLI. For example, if one were to build a private CoreCLR package, on a particular OS (e.g. Windows) and want to test it through, there are no extensibility points to do so.

Wouldn't updating the project.json here: `https://github.com/dotnet/core-setup/blob/master/pkg/projects/Microsoft.NETCore.App/project.json` with whatever private version you wanted and then adding an entry to the directory containing that package to the NuGet.Config work?

**Comment by eerhardt** [comment]
> Wouldn't updating the project.json here: https://github.com/dotnet/core-setup/blob/master/pkg/projects/Microsoft.NETCore.App/project.json with whatever private version you wanted and then adding an entry to the directory containing that package to the NuGet.Config work?

The issue we encountered over and over in the CLI repo was that NuGet attempted to restore for all RIDs specified in the "runtimes" section of a project.json.  Thus, you needed a private version for every RID we support, or else your restore would fail.

To solve that, we generated stub/fake packages for the DotNetHost* packages for the RIDs we didn't have. Here, we would need to create stub/fake packages for all 'runtime' packages for the private version you wanted to test.

See https://github.com/NuGet/Home/issues/2534 for discussion on this issue.
